File: snippets/fsharp/System/Type/GetConstructors/source2.fs
module source2

// <Snippet2>
open System.Reflection

type t() =
    static do ()
    new (i: int) = t ()

let p = typeof<t>.GetConstructors(BindingFlags.Public ||| BindingFlags.Static ||| BindingFlags.NonPublic ||| BindingFlags.Instance)
printfn $"{p.Length}"

for c in p do
    printfn $"{c.IsStatic}"
// </Snippet2>

File: snippets/csharp/System/Type/GetConstructors/source2.cs
// <Snippet2>
using System;
using System.Reflection;

public class ConstructorSample2
{
    public ConstructorSample2() { }
    static ConstructorSample2() { }
    public ConstructorSample2(int i) { }
    public static void Run()
    {
        ConstructorInfo[] p = typeof(ConstructorSample2).GetConstructors(
           BindingFlags.Public | BindingFlags.Static |
           BindingFlags.NonPublic | BindingFlags.Instance);
        Console.WriteLine(p.Length);

        for (int i = 0; i < p.Length; i++)
        {
            Console.WriteLine(p[i].IsStatic);
        }
    }
}
// </Snippet2>
